src/cephadm/cephadm

```diff
@@ -4260,6 +4264,11 @@ def _pull_image(ctx, image, insecure=False):
         if not ret:
             return
 
+        err_str = ''
+        if 'unauthorized' in err:
+            err_str = 'Failed to pull container image. Check that correct registry credentials are provided by --registry-url, --registry-username, --registry-password, or supply --registry-json with credentials'
```

Comment:
I like raising this special error type here, but I don't think the current code solves the issue I mentioned previously that if `_pull_image` is called from somewhere that isn't bootstrap, the advice won't make sense. Maybe instead of just raising this special error type with a certain message in all cases we can catch it in all calls to   `_pull_image` and then just craft an error message based on which call (there's only 3 places we call this so I think it would be possible). Basically, this section here in `_pull_image` would just be something like
```
if 'unauthorized' in err:
    raise UnauthorizedRegistryError()
```
then in the bootstrap section where we call `_pull_image`
```
except UnauthorizedRegistryError as e:
    raise Error("<your-error-message-you-have-here>")
```
then in command_adopt and command_pull it would be done similarly except the message is different. For adopt we probably want to recommend using `cephadm registry-login` (adopt command does not have the registry info flags itself, so they would need to explicitly use the login command) and for `command_pull` since that could reasonably be run from both the mgr module or directly with the cephadm binary maybe just a more generic message saying to make sure you're host(s) are logged into the registry.

Thoughts?